### PR TITLE
Benji add trimet provider

### PR DESCRIPTION
**Pull request by sidetrackedmind**
## Added a new provider "gtfs-trimet". 
Instead of modifying "gtfs-realtime" I made a new "gtfs-trimet" to work with the trimet feed. Some major difference between "gtfs-trimet" and the general "gtfs-realtime":

- "gtfs-realtime" returns data in a compressed format (I think it might be protobuf) vs. "gtfs-trimet" returns data as json
-  "gtfs-realtime" vehicle response schema is nested vs. "gtfs-trimet" returns unnested json

## Update schema

- Changed everything to camel case. For example, `tripId` and `RouteId`. 
- Matched "status" to "messageCode" in the trimet response
- Matched "stopIndex" to "nextStopSeq" in the trimet response

Final schema looks like this
image

## Future development
I'm not sure what will happen if the opentransit-collect encounters bad json. I'm hoping it errors out that attempt but tries again in 15 seconds. Worst case, it errors and stops the docker container. If this happens, I'll have to restart the container on my raspberry pi. It would be better to build in some resiliency given that we know there are cases when trimet's api sends bad json in the response.

### CODE REVIEW

**Review by youngj — COMMENTED** [review]
What URL are you using for Trimet's GTFS realtime API? I have been using this config which seems to still work with the existing gtfs-realtime provider:

```json
{
  "id": "trimet",
  "provider": "gtfs-realtime",
  "gtfs_realtime_url": "https://developer.trimet.org/ws/V1/VehiclePositions?appID=D250D757BEDC0CC1DE086C2B4"
}
```

**Comment by sidetrackedmind** [comment]
Ah - interesting! Here's what I've been using:
```
{
   "id": "trimet",
   "provider": "gtfs-trimet",
   "gtfs_realtime_url": "https://developer.trimet.org/ws/v2/vehicles?appID=AB1730E42229AC8899E1A84BC"
   }
```
Checking Trimet's developer website again I'm following the info on this vehicle_location specific page: https://developer.trimet.org/ws_docs/vehicle_locations_ws.shtml it shows v2: 
```
**Vehicle location service**
Base HTTP URL: http://developer.trimet.org/ws/v2/vehicles
Base HTTPS URL: https://developer.trimet.org/ws/v2/vehicles 
```
If `v1` still works, I'm okay using it but I'm worried it will be deprecated without notice. What do you think? We could ask trimet (there's a support email) to get a clarification on using v1 vs. v2.

**Comment by youngj** [comment]
https://developer.trimet.org/GTFS.shtml lists http://developer.trimet.org/ws/V1/VehiclePositions as the current GTFS-realtime Vehicle Position feed. I don't think https://developer.trimet.org/ws_docs/vehicle_locations_ws.shtml is actually a GTFS-realtime feed (the GTFS-realtime specification is documented at https://developers.google.com/transit/gtfs-realtime/reference). It seems like this might be a Trimet-specific API although I'm not sure. Since eventually we want OpenTransit to support a large number of transit agencies, it would be better not to have to maintain agency-specific APIs unless that is the only API available.

**Comment by sidetrackedmind** [comment]
Weird that they maintain both. I agree, if the GTFS-realtime standard one is not going away, we should use that.

**Comment by youngj** [comment]
It looks like both the GTFS-realtime Vehicle Position feed and the Vehicle Location Service were added to their website sometime between March and July 2014:

https://web.archive.org/web/20140325193959/https://developer.trimet.org/GTFS.shtml
https://web.archive.org/web/20140704071344/https://developer.trimet.org/GTFS.shtml

**Comment by sidetrackedmind** [comment]
Got it - so safe to say they're supporting both (or have been for many years). Also, if they could only pick one, they would probably pick the GTFS-realtime to support because it (I assume) funnels into Google Maps, etc.

Here's my plan.

- I can go back delete the new provider
- switch the URL in my config
- update the schema of the `gtfs-realtime` provider

After that I'll restart the docker on my raspberry pi using `gtfs-realtime`.

File: src/index.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ const providerNames = [
     'nextbus',
     'marin',
     'gtfs-realtime',
+    'gtfs-trimet',
```
Comment (youngj) [inline src/index.js:36]: Remove deleted provider from list

**Comment by sidetrackedmind** [comment]
Deleted `gtfs-trimet` provider and updated a few items on `gtfs-realtime` to be camel case consistent. See snapshot below of the schema diff:
image

**Review by youngj — APPROVED** [review]
The new property names look good to me. 

I think the existing "marin" provider can also be safely removed because Marin Transit  offers a GTFS-RT VehiclePositions feed (https://marintransit.org/developers).

The Nextbus provider would also need the same property name changes. The Nextbus provider is also broken because the domain of the API server has changed from http://webservices.nextbus.com to https://retro.umoiq.com . Feel free to update that in this PR or we can add a separate issue for it. I think the Nextbus provider is still necessary for Portland Streetcar (https://retro.umoiq.com/service/publicJSONFeed?command=vehicleLocations&t=0&a=portland-sc).

**Comment by youngj** [comment]
If you want to collect data for Portland Streetcar, you can use this config:

```json
   {
     "id": "portland-sc",
     "provider": "nextbus",
     "nextbus_agency_id": "portland-sc"
   },
```

**Pull request merged**